aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -67,18 +67,12 @@ Tensor to(
   bool copy,
   c10::optional<c10::MemoryFormat> optional_memory_format
 ) {
-  // See [Note: hacky wrapper removal for TensorOptions]
-  TensorOptions options_ = TensorOptions().dtype(dtype).layout(layout).device(device).pinned_memory(pin_memory);
-
-  TORCH_CHECK(
-    !(options_.has_memory_format() && optional_memory_format.has_value()),
-    "Cannot set memory_format both in TensorOptions and explicit argument; please delete "
-    "the redundant setter.");
-  auto options = options_.merge_memory_format(optional_memory_format);
-
-  TORCH_CHECK(options.requires_grad_opt() == c10::nullopt,
```

Comment:
From the error message, I guess this should be checking `optional_memory_format.value().requires_grad_opt()` ?

Reply:
My take on this is: once upon a time ago, the `Tensor::to` functions used to accept TensorOptions. Because of that, `Tensor::to` needed to check if `options.requires_grad` is unset because it doesn't support `requires_grad` being set. However, since `Tensor::to` no longer accepts TensorOptions (see signature), it doesn't need to check for requires_grad anymore.

Reply:
Ho I misread the last arg as an optional option. These namings :p